Shared/Samples/Filter features in scene/FilterFeaturesInSceneView.swift

```diff
@@ -0,0 +1,241 @@
+                    id: .osmTopographic
+                )
+            )
+            let basemap = Basemap()
+            basemap.addBaseLayers([vectorTiledLayer, osmBuildings])
+            scene = Scene(basemap: basemap)
+            
+            // Create scene topography.
+            let elevationServiceURL = URL(string: "https://elevation3d.arcgis.com/arcgis/rest/services/WorldElevation3D/Terrain3D/ImageServer")!
+            let elevationSource = ArcGISTiledElevationSource(url: elevationServiceURL)
+            let surface = Surface()
+            surface.addElevationSource(elevationSource)
+            scene.baseSurface = surface
+            
+            // Set the initial viewpoint of the scene.
+            scene.initialViewpoint = .sanFranciscoBuildings
+        }
+        
+        /// Creates a polygon that represents the detailed buildings scene layer full extent.
+        /// - Returns: A polygon.
+        private static func makeFilteringPolygon() -> ArcGIS.Polygon {
+            // The buildings scene layer fullExtent.
+            let extent = Envelope(
+                xRange: -122.514 ... -122.357,
+                yRange: 37.705...37.831,
+                zRange: -148.843...551.801,
+                spatialReference: SpatialReference(wkid: .wgs84, verticalWKID: WKID(5773)!)
+            )
+            
+            let builder = PolygonBuilder(spatialReference: extent.spatialReference)
+            builder.add(Point(x: extent.xMin, y: extent.yMin))
+            builder.add(Point(x: extent.xMax, y: extent.yMin))
+            builder.add(Point(x: extent.xMax, y: extent.yMax))
+            builder.add(Point(x: extent.xMin, y: extent.yMax))
+            
+            return builder.toGeometry()
+        }
+        
+        /// Handles the filter state of the sample by either loading, filtering, or reseting the scene.
+        /// - Parameter filterState: The filter state of the sample.
+        func handleFilterState(_ filterState: FilterState) {
+            switch filterState {
+            case .load:
+                // Show the detailed buildings scene layer and extent graphic.
+                loadScene()
+            case .filter:
+                // Hide buildings within the detailed building extent so they don't clip.
+                filterScene()
+            case .reset:
+                // Reset the scene to its original state.
+                resetScene()
+            }
+        }
+        
+        /// Loads the detailed buildings scene layer and adds an extent graphic.
+        private func loadScene() {
+            scene.addOperationalLayer(buildingsSceneLayer)
+            graphicsOverlay.addGraphic(sanFranciscoExtentGraphic)
+        }
+        
+        /// Applies a polygon filter to the open street map buildings layer.
+        private func filterScene() {
+            // Initially, the building layer does not have a polygon filter, set it.
+            if osmBuildings.polygonFilter == nil {
```

Comment:
This condition can be changed into sth like `if let polygonFilter = osmBuildings.polygonFilter { } else { }`, so the optional chaining can be avoided.